### DISCUSSION TITLE
Replacement of shapes.basic.Generic in JointJS 4?

### DISCUSSION THREAD

**Discussion by JanTheMan1010** [clientIO/joint#2760] [Q&A] [answered]
### Introduction

In JointJS 3.x existed shapes.basic.Generic, as a generic shape you can extend.

I haven't found anything simliar in JointJS 4.

What can I use instead?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
`joint.shapes.basic` were removed completely in favor of [joint.shapes.standard](https://docs.jointjs.com/api/shapes/standard/) shapes.

Release notes: https://docs.jointjs.com/learn/release-notes/4.0.0#shapes-basic-remove

**Reply by JanTheMan1010** [reply]
Thank you very much for the response. I knew about joint.shapes.standard shapes already.

I desperately try to find a "base model" like shapes.basic.Generic was.

From my understanding this "base model" can be found in your current version 4.0.4* starting in line 20513:
```
// Element base model.
// -----------------------------

var Element$1 = Cell.extend({
   defaults: {
	        position: { x: 0, y: 0 },
   // ...
  }
   // ...
});

```

From my understanding Element$1 was basically shapes.basic.Generic of version 3.7.x. I understand it's removed now.

Is there a way to build custom shapes based on Element$1 base model in version 4?

* https://cdn.jsdelivr.net/npm/@joint/core@4.0.4/dist/joint.js

**Reply by kumilingus** [reply, marked as answer]
It's `dia.Element`. Here's the tutorial on [custom shapes](https://docs.jointjs.com/learn/features/shapes/custom-shapes/).
